dan_layer/core/src/services/signing_service.rs

```diff
@@ -39,16 +49,57 @@ impl NodeIdentitySigningService {
     pub fn new(node_identity: Arc<NodeIdentity>) -> Self {
         Self { node_identity }
     }
+
+    fn create_challenge(public_key: &PublicKey, public_nonce: &PublicKey, challenge: &[u8]) -> FixedHash {
+        tari_hasher::<TariDanCoreHashDomain>("signing_service")
+            .chain(&key_to_fixed_bytes(public_key))
+            .chain(&key_to_fixed_bytes(public_nonce))
+            .chain(challenge)
+            .result()
+    }
 }
 
 impl SigningService for NodeIdentitySigningService {
-    fn sign(&self, _challenge: &[u8]) -> Result<ValidatorMetadata, DigitalAssetError> {
-        // TODO better sig
-        Ok(ValidatorMetadata {
-            public_key: Vec::from(self.node_identity.public_key().as_bytes()),
-            signature: Vec::new(),
-            merkle_proof: Vec::new(),
-            merkle_leaf_index: 0,
-        })
+    fn sign(&self, challenge: &[u8]) -> Option<Signature> {
+        let (nonce, public_nonce) = create_key_pair();
+        let challenge = Self::create_challenge(&public_nonce, self.node_identity.public_key(), challenge);
+        let sig = Signature::sign(self.node_identity.secret_key().clone(), nonce, &*challenge).ok()?;
```

Comment:
todo: change to sign_raw when tari_crypto is updated

Reply:
Yeah I want to update all the tari deps soon